dboperations/dboperations.go

```diff
@@ -0,0 +1,180 @@
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+	"github.com/chef/omnitruck-service/middleware/db"
+	"github.com/chef/omnitruck-service/models"
+)
+
+const (
+	SKU_PARTITION_KEY     = "sku"
+	PRODUCT_PARTITION_KEY = "product"
+	PRODUCT_SORT_KEY      = "version"
+)
+
+type IDbOperations interface {
+	GetPackages(partitionValue string, sortValue string) (models.ProductDetails, error)
+	GetVersionAll(partitionValue string) ([]string, error)
+	GetMetaData(partitionValue string, sortValue string, platform string, platformVersion string, architecture string) (models.ProductDetails, error)
+	GetVersionLatest(partitionValue string) (string, error)
+	GetRelatedProducts(partitionValue string) (models.Sku, error)
+}
+
+type IDynamoDBOps interface {
+	GetItem(input *dynamodb.GetItemInput) (*dynamodb.GetItemOutput, error)
+	Scan(input *dynamodb.ScanInput) (*dynamodb.ScanOutput, error)
+}
+
+type DbOperationsService struct {
+	db               IDynamoDBOps
+	productTableName string
+	skuTableName     string
+}
+
+func NewDbOperationsService(dbConnection dbconnection.DbConnection) *DbOperationsService {
+	return &DbOperationsService{
+		db:               dbConnection.GetDbConnection(),
+		productTableName: os.Getenv("PRODUCT_TABLE_NAME"),
+		skuTableName:     os.Getenv("SKU_TABLE_NAME"),
+	}
+}
+
+func (dbo *DbOperationsService) GetPackages(partitionValue string, sortValue string) (models.ProductDetails, error) {
+	log.Println(dbo.productTableName)
+	input := &dynamodb.GetItemInput{
+		TableName: aws.String(dbo.productTableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			PRODUCT_PARTITION_KEY: {S: aws.String(partitionValue)},
+			PRODUCT_SORT_KEY:      {S: aws.String(sortValue)},
+		},
+	}
+	res, err := dbo.db.GetItem(input)
+	if err != nil {
+		log.Println("error while using GetItem:", err)
+		return models.ProductDetails{}, err
+	}
+	var response models.ProductDetails
+	if err := dynamodbattribute.UnmarshalMap(res.Item, &response); err != nil {
+		return models.ProductDetails{}, err
+	}
+	return response, nil
+}
+
+func (dbo *DbOperationsService) GetVersionAll(partitionValue string) ([]string, error) {
+	res, err := dbo.fetchDataValues(partitionValue, dbo.productTableName)
+	if err != nil {
+		log.Printf("error in getting the Database value: %v", err)
+		return nil, err
+	}
+	var response models.ProductDetails
+	versionsArray := []string{}
+	for _, i := range res.Items {
+		err = dynamodbattribute.UnmarshalMap(i, &response)
+		versionsArray = append(versionsArray, response.Version)
+		if err != nil {
+			log.Printf("Got error unmarshalling: %s", err)
+			return nil, err
+		}
+
+	}
+	return versionsArray, nil
+}
+
+func (dbo *DbOperationsService) GetMetaData(partitionValue string, sortValue string, platform string, platformVersion string, architecture string) (models.ProductDetails, error) {
```

Comment:
func should return models.Metadata of where arch,platform,platform version matched instead of models.ProductDetails

Reply:
Updated the response type as the metaData struct rather then product Details struct